### PR TITLE
Remove some dependencies

### PR DESCRIPTION
**Pull request by hrxi**
Specialize `vrf::AliasMethod` to `u64` for this, also decreasing code size.

### CODE REVIEW

File: vrf/src/alias.rs [hunk 1]
```diff
@@ -1,31 +1,13 @@
...
-        + Sub<P>
-        + Mul<P>
-        + FromPrimitive
-        + ToPrimitive
-        + PartialOrd<P>
-        + Ord,
-{
+pub struct AliasMethod {
     /// The total probability - since we work with integers, this is not 1.0, but corresponds to
     /// a the probability 1.0
```
Comment (jsdanielh) [inline vrf/src/alias.rs]:
```suggestion
    /// the probability of 1.0
```

File: vrf/src/alias.rs [hunk 2]
```diff
@@ -124,7 +93,7 @@ where
         self.n == 0
     }
 
-    pub fn total(&self) -> P {
+    pub fn total(&self) -> u64 {
```
Comment (jsdanielh) [inline vrf/src/alias.rs]: Could you add some `rustdoc`?

File: vrf/src/alias.rs [hunk 3]
```diff
@@ -1,31 +1,13 @@
...
-        + Unsigned
-        + Add<P>
-        + Sub<P>
-        + Mul<P>
-        + FromPrimitive
-        + ToPrimitive
-        + PartialOrd<P>
-        + Ord,
-{
+pub struct AliasMethod {
```
Comment (jsdanielh) [inline vrf/src/alias.rs]: Could you add some `rustdoc`?

File: primitives/src/coin.rs [hunk 4]
```diff
@@ -64,6 +61,14 @@ impl Coin {
         *self = self.safe_sub(rhs)?;
         Ok(())
     }
+
+    pub fn saturating_add(self, rhs: Coin) -> Coin {
+        Coin(self.0.saturating_add(rhs.0))
+    }
+
+    pub fn saturating_sub(self, rhs: Coin) -> Coin {
```
Comment (jsdanielh) [inline primitives/src/coin.rs:99]: I know it's self explanatory, but could you add a small `rustdoc`?

File: primitives/src/coin.rs [hunk 5]
```diff
@@ -64,6 +61,14 @@ impl Coin {
         *self = self.safe_sub(rhs)?;
         Ok(())
     }
+
+    pub fn saturating_add(self, rhs: Coin) -> Coin {
```
Comment (jsdanielh) [inline primitives/src/coin.rs:93]: I know it's self explanatory, but could you add a small `rustdoc`?

File: vrf/src/alias.rs (hunk 1 above)
Comment (hrxi) [reply vrf/src/alias.rs]: Done.

File: vrf/src/alias.rs (hunk 2 above)
Comment (hrxi) [reply vrf/src/alias.rs]: Removed the method, it seemed useless.

File: vrf/src/alias.rs (hunk 3 above)
Comment (hrxi) [reply vrf/src/alias.rs]: Done, and renamed it to something clearer to me.

File: primitives/src/coin.rs (hunk 4 above)
Comment (hrxi) [reply primitives/src/coin.rs:99]: Done.

File: primitives/src/coin.rs (hunk 5 above)
Comment (hrxi) [reply primitives/src/coin.rs:93]: Done, and to the rest of the file.

File: vrf/src/discrete.rs [hunk 6]
```diff
@@ -116,34 +102,19 @@ where
         Self { T, n, K, U }
     }
 
+    /// Number of different events.
     pub fn len(&self) -> usize {
         self.n
     }
```
Comment (jsdanielh) [inline vrf/src/discrete.rs]: Clippy is warning that there is not `is_empty` method

Comment (hrxi) [reply vrf/src/discrete.rs]: Removed the `len` method as well.

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**